src/gsi/cads.f90

```diff
@@ -0,0 +1,2064 @@
+         471,  473,  475,  477,  479,  481,  483,  485,  487,  489, &
+         491,  493,  495,  497,  499,  501,  503,  505,  507,  509, &
+         511,  513,  515,  517,  519,  521,  523,  525,  527,  529, &
+         531,  533,  535,  537,  539,  541,  543,  545,  547,  549, &
+         551,  553,  555,  557,  559,  561,  563,  565,  567,  569, &
+         571,  573,  575,  577,  579,  581,  583,  585,  587,  589, &
+         591,  593,  595,  597,  601,  603,  605,  607,  609,  611, &
+         613,  615,  617,  619,  621,  623,  625,  627,  629,  631, &
+         633,  635,  637,  639,  641,  643,  645,  647,  649,  651, &
+         653,  655,  657,  659,  661,  663,  665,  667,  669,  681, &
+         693,  699,  703,  707,  711,  715,  719,  723,  727,  731, &
+         737,  741,  745,  749,  753,  757,  761,  771,  777,  807, &
+         813,  819,  827,  831,  851,  855,  863,  867,  889,  913, &
+        1029, 1091, 1103, 1131, 1141, 1145, 1291, 1323, 1335, 1511, &
+        1733, 1841, 2053, 2179, 2265, 2381, 2387, 2541, 3609, 3767, &
+        3891, 3981, 4187, 4477 /)
+
+      N__GradChkInterval(:) = 0
+      N__GradChkInterval(1:N__Num_Bands) = (/ 25 /)
+
+      N__Window_Width(:) = 0
+      N__Window_Width(1:N__Num_Bands) = (/ 20 /)
+
+      N__Window_Bounds(:,:) = 0
+      N__Window_Bounds(1,1) = 1145
+      N__Window_Bounds(1,2) = 4477
+
+      R__BT_Threshold(:) = 0.0_r_kind
+      R__BT_Threshold(1:N__Num_Bands) = (/ 0.27_r_kind /)
+
+      R__Grad_Threshold(:) = 0.0_r_kind
+      R__Grad_Threshold(1:N__Num_Bands) = (/ 0.02_r_kind /)
+
+      R__Window_Grad_Threshold(:) = 0.0_r_kind
+      R__Window_Grad_Threshold(1) = 0.4_r_kind
+
+      L__Do_Quick_Exit = .TRUE.
+
+
+    ! This is cross-band:
+
+      L__Do_CrossBand = .TRUE.
+
+      N__BandToUse(:) = 0
+      N__BandToUse(1:N__Num_Bands) = (/ 1 /)
+
+    ! This is the setup for imager cloud detection
+
+      L__Do_Imager_Cloud_Detection = .FALSE.
+
+      N__Num_Imager_Chans = 0
+      N__Num_Imager_Clusters = 0
+      N__Imager_Chans(:) = 0
+
+      R__Stddev_Threshold(:) = 0.0_r_kind
+      R__Coverage_Threshold = 0.0_r_kind
+      R__FG_Departure_Threshold = 0.0_r_kind
+
+
+!    CASE DEFAULT
+!      CYCLE
+    END SELECT
+
+  !------------------------------------------------------------------
+  ! Open and read file containing cloud detection setup for the
+  ! current instrument
+  !------------------------------------------------------------------
+
+  INIU1=107
+  OPEN(INIU1,STATUS='OLD',FORM='FORMATTED', &
+       FILE=TRIM(CL__Cloud_Detection_File), IOSTAT=IOS)
+  IF (IOS == 0) THEN
+    READ(INIU1,nml=Cloud_Detect_Coeffs,IOSTAT=IOS)
+    IF (IOS /= 0) THEN
+      CALL CADS_Abort('PROBLEM READING '//TRIM(CL__InstrumentName)//&
+                     'CLOUD DETECTION FILE')
+!    ELSE
+!      WRITE(*,'(3X,A)') TRIM(CL__InstrumentName) // &
+!           ' CLOUD DETECTION FILE READ OK'
+    ENDIF
+  ELSE
+    WRITE(*,'(3X,A)') 'NO '//TRIM(CL__InstrumentName) // &
```

Comment:
Limit this printout to a single task or only execute when `verbose=.true.`

Reply:
While [`1196943`](https://github.com/NOAA-EMC/GSI/pull/616/commits/1196943318f109734e03078b4cf8bf8aa9ea082d) places the CADS printout inside a an `if verbose` block, the subroutine call to `CADS_Setup_Cloud` is placed before `gsi.x` reads the `SETUP` namelist.   See the following code from `gsimod.F90`
```
  call gsi_nstcoupler_init_nml
  call init_radaruse_directDA
  call CADS_Setup_Cloud

 if(mype==0) write(6,*)' at 0 in gsimod, use_gfs_stratosphere,nems_nmmb_regional = ', &
                       use_gfs_stratosphere,nems_nmmb_regional


! Read user input from namelists.  All processor elements                                                                                               
! read the namelist input.  SGI MPI FORTRAN does not allow                                                                                              
! all tasks to read from standard in (unit 5).  Hence, open                                                                                             
! namelist to different unit number and have each task read                                                                                             
! namelist file.                                                                                                                                        
#ifdef ibm_sp
! Initialize table of instruments and data types                                                                                                        
  read(5,setup)
```
Thus, the default value of `verbose=.false.` is always used.  We do not see CADS printout even when `verbose=.true.`

The call to `CADS_Setup_Cloud` needs to be after namelist `SETUP` is read if we want `verbose=.true.` to toggle on CADS printout.